### DISCUSSION TITLE
How can I improve the speed of reading npy files?

### DISCUSSION THREAD

**Discussion by RizerKayseri** [pyodide/pyodide#4449] [Q&A]
On my computer, it takes 6ms to read the contents of a 500MB npy file using the memap mode. But in the virtual environment, the reading process is extended to 300ms, which is excessively long. How can I improve the speed of reading npy files? How can I efficiently read npy files in a node environment, preferably using memory mapping to save some memory?

**Comment by gabrielfougeron** [comment]
You might be interested in this project: https://github.com/aplbrain/npyjs
